pkg/controller/controlplane/valuesprovider.go

```diff
@@ -587,6 +625,74 @@ func getCRCChartValues(
 	return values, nil
 }
 
+// getIPAMChartValues collects and returns the ipam-controller chart values.
+func getIPAMChartValues(
+	cp *extensionsv1alpha1.ControlPlane,
+	cluster *extensionscontroller.Cluster,
+	checksums map[string]string,
+	scaledDown bool,
+) (map[string]interface{}, error) {
+
+	mode := "ipv4"
+	if cluster.Shoot.Spec.Networking != nil &&
+		len(cluster.Shoot.Spec.Networking.IPFamilies) == 1 &&
+		cluster.Shoot.Spec.Networking.IPFamilies[0] == v1beta1.IPFamilyIPv6 {
+		mode = "ipv6"
+	}
+	if cluster.Shoot.Spec.Networking != nil &&
+		len(cluster.Shoot.Spec.Networking.IPFamilies) == 2 {
+		mode = "dual-stack"
+	}
+
+	nodeCidrMaskSizeIPv4 := int32(24)
+	nodeCidrMaskSizeIPv6 := int32(64)
+	if cluster.Shoot.Spec.Kubernetes.KubeControllerManager != nil && cluster.Shoot.Spec.Kubernetes.KubeControllerManager.NodeCIDRMaskSize != nil {
+		if cluster.Shoot.Spec.Networking != nil &&
+			len(cluster.Shoot.Spec.Networking.IPFamilies) == 1 &&
+			cluster.Shoot.Spec.Networking.IPFamilies[0] == v1beta1.IPFamilyIPv4 {
+			nodeCidrMaskSizeIPv4 = *cluster.Shoot.Spec.Kubernetes.KubeControllerManager.NodeCIDRMaskSize
+		}
+		if cluster.Shoot.Spec.Networking != nil &&
+			len(cluster.Shoot.Spec.Networking.IPFamilies) == 1 &&
+			cluster.Shoot.Spec.Networking.IPFamilies[0] == v1beta1.IPFamilyIPv6 {
+			nodeCidrMaskSizeIPv6 = *cluster.Shoot.Spec.Kubernetes.KubeControllerManager.NodeCIDRMaskSize
+		}
+	}
+
+	podNetwork := "192.168.0.0/16"
```

Comment:
Can you maybe explain where this value is coming from and why it is hardcoded here ? Shouldn't we return an error if it's missing ?

Reply:
The ipam controller requires a value for the podNetwork. In case of IPv6 you do not want to specify it and take a default value. I took the same value as specified in the ccm for the podNetwork, see: https://github.com/gardener/gardener-extension-provider-aws/blob/180b867ff68470bae677d1d4c2fe097c6949e043/charts/internal/seed-controlplane/charts/cloud-controller-manager/values.yaml#L4